src/build_vehicle_gui.cpp

```diff
@@ -1101,8 +1462,20 @@ struct BuildVehicleWindow : Window {
 		} else {
 			this->SelectEngine(INVALID_ENGINE);
 		}
+		InvalidateData();
 	}
 
+
+	const VehicleFilter& getFilter() const {
```

Comment:
All function names should start with capital letter. (Same for `setFilter` below.)